### PR TITLE
creating neural network class with constuctor

### PR DESCRIPTION
**Pull request by arhamahmed**

### CODE REVIEW

**Comment by irfansharif** [comment]
ah, missed this earlier. i restructured from the default setup.

File: lib/cerebrum.rb [hunk 1]
```diff
@@ -1,5 +1,47 @@
 require "cerebrum/version"
+require 'matrix'
 
 module Cerebrum
```
Comment (irfansharif) [inline lib/cerebrum.rb:4]: look at the latest master, this was changed to `class`. the way it is right here makes the API call resolve to `Cerebrum::NeuralNetwork.new`, which would be fine but we're not really using the `Cerebrum` namespace for anything else.

File: lib/cerebrum.rb [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+  class NeuralNetwork
+    def initialize(options)
+      options = options || {}
+      learningRate = options.learningRate || 0.3
+      momentum = option.momentum || 0.1
+      hiddenSizes = options.hiddenLayers
+      binaryThresh = options.binaryThresh || 0.5
+    end
+
+    def initializeNN(sizes)
```
Comment (irfansharif) [inline lib/cerebrum.rb:14]: was this supposed to be the port to the original `train` method?

**Comment by arhamahmed** [comment]
no, it's a port of the class constructor

**Comment by irfansharif** [comment]
i'll close this and add it separately through another.

File: lib/cerebrum.rb [hunk 3]
```diff
@@ -1,5 +1,47 @@
 require "cerebrum/version"
+require 'matrix'
```
Comment (irfansharif) [inline lib/cerebrum.rb:2]: also ps, when adding external dependencies you'll need to update the Gemfile so anyone can just `bundle install` and it'll pick up what they're missing. this is equivalent to `package.json` and `npm install`

**Pull request closed without merging**